scripts/mysql_system_tables_fix.sql

```diff
@@ -1002,62 +1002,213 @@ INSERT IGNORE INTO mysql.global_grants VALUES ('mysql.session', 'localhost', 'SE
 # upgrade.
 INSERT IGNORE INTO mysql.global_grants VALUES ('mysql.session', 'localhost', 'SYSTEM_USER', 'N');
 
+set @is_mysql_encrypted = (select ENCRYPTION from information_schema.INNODB_TABLESPACES where NAME='mysql');
+
+SET @str="ALTER TABLE mysql.db ENCRYPTION='Y'";
+SET @cmd = IF(STRCMP(@is_mysql_encrypted,'Y'), 'SET @dummy = 0', @str);
+PREPARE stmt FROM @cmd;
+EXECUTE stmt;
+DROP PREPARE stmt;
+
 # Move all system tables with InnoDB storage engine to mysql tablespace.
 SET @cmd="ALTER TABLE mysql.db TABLESPACE = mysql";
```

Comment:
Is it possible to combine ENCRYPTION='Y' and TABLESPACE=mysql ?

Reply:
I do not see a convenient way of combining those two without explicitly specifying ENCRYPTION=N clause also. Also I do not see much benefit in refactoring this.

Reply:
Ok, fine..